reviews/review-01+2020-11-02.md

```diff
@@ -0,0 +1,305 @@
+    sentence, but should happen sometimes before the second or third
+    paragraph) introducing the topic at hand and why we care about it.
+    This formula may not work for every section, but I think it's worth
+    checking.
+
+    -   Good examples:
+
+        -   Introduction to chapter 3 (Going Further with the Unix
+            Shell): "This chapter extends these ideas to show how we can
+            create new tools by saving commands in files and how to use
+            a more powerful version of wildcards to extract data from
+            files."
+
+            -   Topic: We're going to put commands in files and use
+                wildcards
+
+            -   Why: We can create new tools and get data (and do better
+                work)
+
+        -   3.2, using variables in a script: "Getting the name of the
+            author for only one of the books isn't all that useful. What
+            we really want is a way to get the name of the author from
+            any of our files." (and then go on to talk about variables)
+
+            -   Topic: Using variables
+
+            -   Why: We want to solve the problem of analyzing all of
+                our data.
+
+    -   Examples that could be improved:
+
+        -   2.1, starting to use the shell: I'm looking for something
+            like "Our first shell commands will allow us to navigate and
+            view the contents of our file system, a foundational skill
+            for using the shell to run other commands and use files
+            organized in a project."
+
+        -   3.6, introducing shell variables: I'm missing the final "so
+            what"---like "knowing about shell variables allows us to
+            customize our shell working environment if we use it
+            frequently."
+
+    -   If you only review one chapter for these "topic sentences" I
+        would recommend chapter 7. The content in that chapter is
+        crucially important, and currently can read as somewhat
+        unconnected topics around people and project management. A topic
+        sentence in each section that links back to the goal of
+        empowering people to collaborate together and in a safe and
+        productive manner (or connecting a particular section to a
+        previous section in the chapter, e.g. 7.6 is really a follow on
+        from 7.5) would go a long way towards helping that material
+        cohere. That, or adding more details to the introduction,
+        outlining the major topics that will be covered in the chapter.
+
+-   I had the most difficulty in following chapter 10 (Error Handling);
+    it was the one chapter where I felt like I didn't have a sufficient
+    mental model to understand the details of the chapter. I'm not sure
+    whether the solution is for me to have a better understanding of
+    Python or for the chapter to more explicitly generate a mental model
+    for errors and exceptions.
```

Comment:
What do you all think of this, how much should we try to build a mental model for the reader? One thing that comes to mind would be to add another intro paragraph just before the first section that introduces all the sections in 2-3 sentences each and also describe how they are connected. So giving the entire thread of the story up front, which I think would be useful in terms of a high level mental model of the different parts of error handling and this chapter (but maybe not for more technical details). These are the sections:

- Exceptions
- Writing Useful Error Messages
- Kinds of Errors
- Reporting Errors

If we think this is a good idea I can try writing one. Not sure if this is enough and open to hear other ideas.